### PR TITLE
Dragonflight changes

### PR DESCRIPTION
**Pull request by Solanya**
Still running through side quests to make last scaling/anim changes but everything else has been working fine.
The one annoying issue there is right now is down to Blizzard's Model API being exceptionally borked when an NPC has a weapon out, throwing a lot of OnAnimFinished all over the place. Unless I find a way to sheathe/remove weapons from models, that's a few manual animations that have to be put in to ensure animations don't glitch out.

### CODE REVIEW

File: reputation-bar/reputation-bar.lua [hunk 1]
```diff
@@ -16,39 +16,62 @@
...
 local DEFAULT_ICON = [[Interface\Common\friendship-heart]];
 
-hooksecurefunc("NPCFriendshipStatusBar_Update", function(_, factionID --[[ = nil ]])
-	local id, rep, maxRep, name, text, texture, reaction, threshold, nextThreshold = GetFriendshipReputation(factionID);
-	StorylineReputationBar.friendshipFactionID = id;
-	if ( id and id > 0 ) then
-		if ( not nextThreshold ) then
-			threshold, nextThreshold, rep = 0, 1, 1;
+local function OnReputationBarLoad(self)
+	self:SetStatusBarTexture(1, 1, 1, "BORDER", -1);
```
Comment (Meorawr) [inline reputation-bar/reputation-bar.lua:33]: I don't believe this is valid anymore in 10.0; they simplified this API to only accept a single "asset" argument as a string if the autogenerated API documentation is to be believed. I suspect the request as such for the "BORDER" draw layer is ignored, along with the sublevel.

I'd suggest the clearer alternative of manually creating a texture object and assigning it via the SetStatusBarTexture method. The API documentation lies a bit here and does accept this still it seems:

```lua
local texture = self:CreateTexture(nil, "BORDER", nil, -1);
texture:SetGradient("VERTICAL", ...);
self:SetStatusBarTexture(texture);
```

File: reputation-bar/reputation-bar.lua [hunk 2]
```diff
@@ -16,39 +16,62 @@
...
+	self:GetStatusBarTexture():SetGradient("VERTICAL", Ellyb.Color.CreateFromRGBA(8/255, 93/255, 72/255, 1), Ellyb.Color.CreateFromRGBA(11/255, 136/255, 105/255, 1));
+end
+
+local function OnReputationBarEnter(self)
+	ShowFriendshipReputationTooltip(self.friendshipFactionID, self, "ANCHOR_TOPRIGHT")
```
Comment (Meorawr) [inline reputation-bar/reputation-bar.lua:38]: This function was moved to a mixin; `ReputationBarMixin.ShowFriendshipReputationTooltip`.

**Review by Meorawr — COMMENTED** [review]
In addition to this - you _may_ want to consider testing the following events in-place of anything like `GOSSIP_SHOW`/`GOSSIP_CLOSED`:

* `PLAYER_INTERACTION_MANAGER_FRAME_SHOW`
* `PLAYER_INTERACTION_MANAGER_FRAME_HIDE`

When these events fire they'll specify an interaction type in their payload that's a member of the `Enum.PlayerInteractionType` enumeration; eg. `Enum.PlayerInteractionType.Gossip`.

I _think_ the intention going forward is that the old SHOW/HIDE events for interactions will eventually be removed. Many of them already have, but it seems that a few (*including gossip*) remain. Could be something to keep in mind as a possibly breaking change in a future patch.

**Comment by Solanya** [comment]
New gossip events have been put in place, will test more but it seems much better, no awkward window closing and reopening as I was having with the other event.
Also fixed the reputation bar, it now uses the new template and only replaces the OnEnter script (for the tooltip anchor) and SetStatusBarColor to account for the nightfallen custom color (using the faction IDs instead for simplicity)

**Pull request merged**